### DISCUSSION TITLE
Filter top-level item based on values in m2a items

### DISCUSSION THREAD

**Discussion by flicksolutions** [directus/directus#5000] [Archive]
Hello again! As discussed on [Discord](https://discord.com/channels/725371605378924594/741319183060566077/830453169842683915) last week, I'm trying to filter a request in the following pattern:

```
filter: {
  "relation-table": {
    "item":  {
      "id": {
        "_eq": 1
      }
    }
  }
}
```

The field _relation-table_ is the standard connection-table you get, using a m2a-builder. Its fields are _xx-id_, _collection_, _item_.
The response from the Server is a `500`.

When I try the same request in a o2m-relationship works. Therefore, I think this must be a bug.

My current workaround is to replace the m2a-relation with two separate m2o-relations which is suboptimal.

Thanks for your views on that!

---

### Project details
Directus Version: 9.0.0-rc.57
Environment: production
OS: Linux 5.4.0-70-generic
Node: 14.16.0

**Comment by rijkvanzanten** [comment]
Heya! The filter can't dynamically guess what sub-collection you're trying to filter in, so there's no way for it to know where that `id` field is to be found. That being said, the ID itself is stored on the `item` field within `relation-table`, so you should be able to do:

```js
filter: {
  "relation-table": {
    "item":  {
      "_eq": 1
    }
  }
}
```

**Comment by flicksolutions** [comment]
Hey!
Thanks for your answer!
Well, it wouldn't matter on which sub-collection it would filter, since the field exists in all the sub-colletions.
But I think I understand why it's something else than with other relation-types...

yeah, the ID was a suboptimal field as an example.

**Reply by rijkvanzanten** [reply]
> Well, it wouldn't matter on which sub-collection it would filter, since the field exists in all the sub-colletions.

Yeah! That makes sense from a logical / english perspective, but once you convert it into `WHERE <collection>.<field> = x` you start to realize that there's a problem in the executing the SQL side of things 🙂 

I think it might be possible to do some extra magic to do the joins based on the compound related key in order to dynamically construct this query with multiple where / or clauses instead of the single filter, but that's TBD
